cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -92,7 +92,23 @@ type Etcd struct {
 	DataDir   string
 	ExtraArgs map[string]string
 	// Image specifies which container image to use for running etcd. If empty, automatically populated by kubeadm using the image repository and default etcd version
-	Image string
+	Image      string
+	SelfHosted *SelfHostedEtcd
+}
+
+// SelfHostedEtcd describes options required to configure self-hosted etcd
+type SelfHostedEtcd struct {
+	// CertificatesDir represents the directory where all etcd TLS assets are stored. By default this is
+	// a dir names "etcd" in the mains CertificatesDir value.
+	CertificatesDir string
```

Comment:
My understanding was that certificates for etcd are going to be stored in secrets. If this is not true, is it required to move those certs among masters in an HA scenario?

Reply:
They will be, but they need to be generated to a master dir first. Should we generate to a tmp dir or something?

Reply:
if we are going to use a temp dir, we can consider this an internal detail and simplify the API

Reply:
Let's punt on this discussion until we have my follow-up PR which implements this logic. I agree that there might be opportunity to simplify the API (which is preferred)